pyhf/tensor/mxnet_backend.py

```diff
@@ -105,11 +105,17 @@ def astensor(self, tensor_in, dtype='float'):
             MXNet NDArray: A multi-dimensional, fixed-size homogenous array.
         """
         dtypemap = {'float': 'float32', 'int': 'int32', 'bool': 'uint8'}
-        dtype = dtypemap[dtype]
         try:
-            tensor = nd.array(tensor_in, dtype=dtype)
-        except ValueError:
+            dtype = dtypemap[dtype]
+        except KeyError:
+            log.error('Invalid dtype: dtype must be float, int, or bool.')
```

Comment:
If there is a better way to raise the offending error with a custom message that gives more information let me know and I'll implement it.

Reply:
that's fine.  just add a test to ensure coverage.

Reply:
Doesn't the `test_astensor` test I added do that?

Reply:
> Doesn't the `test_astensor` test I added do that?

You don't have full coverage -- e.g. checking for the KeyError message, and so on.

Reply:
I probably don't know how to properly test failure cases in pytest (will go read now) but why doesn't

```
assert tb.astensor([1, 2, 3], dtype='long')
```

in `test_astensor` trigger the `KeyError` case? Or am I totally missing the point here?

Reply:
Ah, did you mean that I should be using [`mark.xfail`'s `raises` argument](https://docs.pytest.org/en/latest/skipping.html#raises-parameter)?

Reply:
Yes, you're right that `pytest.raises` is more appropriate here as from the [pytest docs](https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions)

> Using `pytest.raises` is likely to be better for cases where you are testing exceptions your own code is deliberately raising, whereas using `@pytest.mark.xfail` with a check function is probably better for something like documenting unfixed bugs (where the test describes what "should" happen) or bugs in dependencies.